lib/improver/tests/ensemble_calibration/ensemble_calibration_utilities/test_SplitHistoricForecastAndTruth.py

```diff
@@ -0,0 +1,223 @@
+            self.plugin._find_required_cubes_using_metadata(
+                self.combined, input_dict)
+
+
+class Test__filter_non_matching_cubes(SetupCubesAndDicts):
+    """Test the _filter_non_matching_cubes method."""
+
+    def test_all_matching(self):
+        """Test for when the historic forecast and truth cubes all match."""
+        hf_result, truth_result = self.plugin._filter_non_matching_cubes(
+            self.historic_forecasts, self.truth)
+        self.assertEqual(hf_result, self.historic_forecasts)
+        self.assertEqual(truth_result, self.truth)
+
+    def test_fewer_historic_forecasts(self):
+        """Test for when there are fewer historic forecasts than truths,
+        for example, if there is a missing forecast cycle."""
+        hf_result, truth_result = self.plugin._filter_non_matching_cubes(
+            self.partial_historic_forecasts, self.truth)
+        self.assertEqual(hf_result, self.partial_historic_forecasts)
+        self.assertEqual(truth_result, self.partial_truth)
+
+    def test_fewer_truths(self):
+        """Test for when there are fewer truths than historic forecasts,
+        for example, if there is a missing analysis."""
+        hf_result, truth_result = self.plugin._filter_non_matching_cubes(
+            self.historic_forecasts, self.partial_truth)
+        self.assertEqual(hf_result, self.partial_historic_forecasts)
+        self.assertEqual(truth_result, self.partial_truth)
+
+
+class Test_process(SetupCubesAndDicts):
+    """Test the process method."""
+
+    def test_basic(self):
+        """Test that the input cubelist combining historic forecasts and truth
+        can be split using the metadata dictionaries provided."""
+        hf_result, truth_result = self.plugin.process(self.combined)
+        self.assertEqual(hf_result, self.historic_forecasts_cube)
+        self.assertEqual(truth_result, self.truth_cube)
+
+    def test_fewer_historic_forecasts(self):
+        """Test that the input cubelist combining historic forecasts and truth
+        can be split using the metadata dictionaries provided, when there are
+        fewer historic forecasts than truths."""
+        combined = self.partial_historic_forecasts + self.truth
+
+        hf_result, truth_result = self.plugin.process(combined)
+        self.assertEqual(hf_result, self.partial_historic_forecasts_cube)
+        self.assertEqual(truth_result, self.partial_truth_cube)
+
+    def test_fewer_truths(self):
+        """Test that the input cubelist combining historic forecasts and truth
+        can be split using the metadata dictionaries provided, when there are
+        fewer truths than historic forecasts."""
+        combined = self.historic_forecasts + self.partial_truth
+
+        hf_result, truth_result = self.plugin.process(combined)
+        self.assertEqual(hf_result, self.partial_historic_forecasts_cube)
+        self.assertEqual(truth_result, self.partial_truth_cube)
+
```

Comment:
Likewise, if you accept my point about it being worth always applying the filter, we should have an offset mismatch test here as to ensure it invokes the filtering.

Reply:
As I've now moved the filtering function to the `EstimateCoefficientsForEnsembleCalibration` plugin, I've added a test to the `process` method of this plugin.